### PR TITLE
Fork the inspector panel and hide the forked version behind a feature flag

### PR DESCRIPTION
**Pull request by elliette**
Work towards https://github.com/flutter/devtools/issues/7861

This PR:
- Forks the inspector panel
- Hides the V2 fork behind a new feature flag `inspectorV2`

Currently the only change between the current inspector panel and the forked version is that the forked version has appends "2.0" to the Widget Tree, Layout Explorer, and Widget Details Tree:

Screenshot 2024-05-30 at 1 50 32 PM

Follow up will be forking the inspector tests as well

### CODE REVIEW

File: packages/devtools_app/lib/src/app.dart [hunk 1]
```diff
@@ -594,18 +597,34 @@ List<DevToolsScreen> defaultScreens({
...
+          ),
+          detailsTree: inspector_v2.InspectorTreeController(
+            gaId: InspectorScreenMetrics.detailsTreeGaId,
+          ),
+          treeType: FlutterTreeType.widget,
         ),
-        detailsTree: InspectorTreeController(
-          gaId: InspectorScreenMetrics.detailsTreeGaId,
+      ),
+    if (!FeatureFlags.inspectorV2)
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/app.dart:615]: use a ternary operator to make this more verbose (see example for logging screen below)

File: packages/devtools_app/lib/src/screens/inspector_v2/inspector_breadcrumbs.dart [hunk 2]
```diff
@@ -0,0 +1,209 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/inspector_v2/inspector_breadcrumbs.dart:1]: 2024 for all these forked files

**Review by kenzieschmoll — APPROVED** [review]
lgtm with one comment

File: packages/devtools_app/lib/src/screens/inspector_v2/inspector_breadcrumbs.dart (hunk 2 above)
Comment (elliette) [reply packages/devtools_app/lib/src/screens/inspector_v2/inspector_breadcrumbs.dart]: done!

**Comment by elliette** [comment]
Closing in favor of https://github.com/flutter/devtools/pull/7864 which is identical but preserves the git history of the copied files.

**Pull request closed without merging**